cognite/powerops/case/case.py

```diff
@@ -0,0 +1,111 @@
+from __future__ import annotations
+
+import logging
+import tempfile
+from typing import Any
+
+import yaml
+
+logger = logging.getLogger(__name__)
+
+
+class Case:
+    r"""
+    Wrapper around YAML file for SHOP, describing a case.
+
+    Main features:
+     * Values can be retrieved or set / changed like with a dict.
+        * Dot syntax is also supported.
+     * Can hold references to additional files ("extra files").
+        * These are just paths to files, their content is not accessed here.
+     * Loading from string or file (with `Case.load_yaml(path_to_yaml)`).
+     * Saving to yaml (with `case.save_yaml(path_to_file)`).
+
+    Examples:
+       >>> case = Case('''
+       ... foo:
+       ...   bar1: 11
+       ...   bar2: 22
+       ... ''')
+       >>> case["foo"]["bar2"] = 202
+       >>> case.yaml
+       'foo:\n  bar1: 11\n  bar2: 202\n'
+       >>> case["foo.bar1"] = 101
+       >>> case["foo.bar1"]
+       101
+
+    """
+
+    def __init__(self, data: str = "") -> None:
+        self._data = {}
+        self._extra_files = []
+
+        yaml_docs = yaml.safe_load_all(data)
+        for i, yaml_doc in enumerate(yaml_docs):
+            if not i:
+                self._data = yaml_doc
+            else:
+                tmp_file = tempfile.NamedTemporaryFile(
+                    mode="w",
+                    delete=False,
+                    prefix="powerops-sdk-tmp-",
+                    suffix=".yaml",
+                )
+                tmp_file.write(yaml.dump(yaml_doc))
+                tmp_file.close()
+                self.add_extra_file(tmp_file.name)
+
+    @classmethod
+    def load_yaml(cls, yaml_path: str) -> Case:
+        logger.info(f"loading case file: {yaml_path}")
+        with open(yaml_path, "r") as fh:
```

Comment:
fh? I understand the usage after seeing the syntax, but I would suggest giving it a name that describes the purpose.

Reply:
Old habits die hard... it's short for "file handle". I'll change it.